pkg/alertmanager/operator.go

```diff
@@ -1267,8 +1267,12 @@ func checkDiscordConfigs(
 			return err
 		}
 
-		if _, err := store.GetSecretKey(ctx, namespace, config.APIURL); err != nil {
-			return fmt.Errorf("failed to retrieve API URL: %w", err)
+		url, err := store.GetSecretKey(ctx, namespace, config.APIURL)
+		if err != nil {
+			return err
+		}
+		if _, err := validation.ValidateURL(strings.TrimSpace(url)); err != nil {
+			return fmt.Errorf("discord webhook 'url' %s invalid: %w", url, err)
```

Comment:
The error from ValidateURL() already contains the URL value. Also since we treat the URL as a secret, we shouldn't log it and we'd need a ValidateSecretURL() function.
```suggestion
			return err
```

Reply:
> we'd need a ValidateSecretURL() function

You mean we'd need to define a new ValidateSecretURL() function? Why? I thought we already got the URL string from GetSecretKey() in the previous line, so ValidateURL() is enough? 🤔

Reply:
It's because `ValidateURL()` would return the URL value in the error which isn't something you want for a "secret" URL.